packages/components/skeleton/src/SkeletonItem.tsx

```diff
@@ -0,0 +1,95 @@
+import { ComponentPropsWithoutRef, CSSProperties, forwardRef } from 'react'
+
+import {
+  type SkeletonItemStyleProps,
+  skeletonItemStyles,
+  skeletonLineStyles,
+} from './SkeletonItem.styles'
+
+interface SkeletonItemProps extends ComponentPropsWithoutRef<'div'>, SkeletonItemStyleProps {}
+
+export type SkeletonRectangleProps = Omit<SkeletonItemProps, 'shape'> & {
+  width?: string | number
+  height?: string | number
+}
+export type SkeletonCircleProps = Omit<SkeletonItemProps, 'shape'> & { size: string | number }
+export type SkeletonLineProps = Omit<SkeletonItemProps, 'shape'> & {
+  lines?: number
+  /**
+   * Sets the default alignement.
+   * @default start
+   */
+  alignment?: 'start' | 'center' | 'end'
+  /**
+   * Sets the gaps between group items.
+   */
+  gap?: 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl'
+}
+
+const getSizeValue = (size?: number | string): string | undefined => {
+  if (typeof size === 'number') return `${size}px`
+
+  return size
+}
+
+const SkeletonItem = forwardRef<HTMLDivElement, SkeletonItemProps>(
+  ({ shape, className, ...rest }, forwardedRef) => {
+    return (
+      <div
+        ref={forwardedRef}
+        aria-hidden="true"
+        className={skeletonItemStyles({ shape, className })}
+        {...rest}
+      />
+    )
+  }
+)
+
+export const SkeletonRectangle = ({ width = '100%', height, ...rest }: SkeletonRectangleProps) => {
+  return (
+    <SkeletonItem
+      style={
+        {
+          '--skeleton-rect-width': getSizeValue(width),
+          '--skeleton-rect-height': getSizeValue(height),
+        } as CSSProperties
+      }
+      className="h-[--skeleton-rect-height] w-[--skeleton-rect-width]"
```

Comment:
Maybe use inline styles for this instead of TW classNames ? (like we advise our consumers to do when they use arbitrary value)

Reply:
This is what I did initially. But then, when I wanted to overwrite the value there were priority issues :confused: